Dockerfile

```diff
@@ -1,16 +1,23 @@
-FROM node:21
-WORKDIR /usr/src/app
-
 ARG PORT=5000
 
+FROM node:21-alpine AS builder
+WORKDIR /usr/src/app
 RUN npm install -g pnpm
-
 COPY package*.json ./
-COPY pnpm-lock.yaml ./
```

Comment:
Please keep pnpm-lock.yaml during package installation, it ensures that all package versions are always exactly the same even if package.json allows wiggle room. Sometimes even a patch version can change a small thing that leads to instability